lib/internal/util/inspect.js

```diff
@@ -439,6 +439,82 @@ defineColorAlias('inverse', 'swapColors');
 defineColorAlias('inverse', 'swapcolors');
 defineColorAlias('doubleunderline', 'doubleUnderline');
 
+// TODO(BridgeAR): Consider using a class instead and require the user to instantiate it
+// before using to declare their color support (or use a default, if none
+// provided).
+const colorize = {};
+
+function defineColorGetter({ style, codes, enumerable }) {
+  function color(...strings) {
+    let result = '';
+    const ansiCodes = this.ansiCodes
+    let string = strings[0];
+    for (let i = 1; i < strings.length; i++) {
+      string += ' ' + strings[i];
+    }
+
+    // Fast path
+    const searchCode = ansiCodes.length === 1 ? ansiCodes[0].end : '\u001b[';
+    const ansiCodeStart = string.indexOf(searchCode);
+    if (ansiCodeStart === -1) {
+      for (const code of ansiCodes) {
+        result += code.start;
+      }
+    } else {
+      // Slow path
+      const start = string.slice(0, ansiCodeStart - 1);
+      let middle = string.slice(ansiCodeStart - 1, -4);
+      const end = string.slice(-4);
+
+      for (const code of ansiCodes) {
+        result += code.start;
+        // Continue former colors by finding end points and continuing from there.
+        middle = middle.replaceAll(code.end, `$&${code.start}`)
+      }
+      string = `${start}${middle}${end}`;
+    }
+
+    result += string;
+    for (let i = ansiCodes.length - 1; i >= 0; i--) {
+      result += ansiCodes[i].end;
+    }
+
+    return result;
+  }
+
+  Object.defineProperty(colorize, style, {
+    __proto__: null,
+    get: function () {
+      if (typeof this === 'function') {
+        this.ansiCodes.push(codes);
+        return this;
+      }
+      const ansiCodes = [codes];
+      const context = {
+        ansiCodes
+      };
+      let boundColor = color.bind(context);
+      // Enable chaining.
+      Object.setPrototypeOf(boundColor, colorize);
+      boundColor.ansiCodes = ansiCodes;
+      return boundColor;
+    },
+    enumerable,
+  });
+}
+
+for (const [style, descriptor] of Object.entries(Object.getOwnPropertyDescriptors(inspect.colors))) {
```

Comment:
Do we need to use primordials here? @aduh95

Reply:
it's not as important at the top level, but it'd probably be a good idea for consistency